solutions/arrays/matrix-diagonal.js

```diff
@@ -0,0 +1,50 @@
+let fs = require("fs");
+let data = fs.readFileSync(0, 'utf-8');
+let idx = 0;
+data = data.split('\n');
+
+function readLine() {
+  idx++;
+  return data[idx - 1];
+}
+
+// -------- Do NOT edit anything above this line ----------
+
+// name your function as change_diagonal and it should take list as input
+function change_diagonal(arr)
+{
+	for(let row=0;row<arr.length;row++)
+	{
+		for(let column=0;column<arr.length;column++)

```

Comment:
@DevratAgarwal This is O(n^2) time. We don't need a nested loop. We can do in O(n) time.